src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -293,14 +298,13 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
                             if (refreshAll)
                             {
                                 // Trigger a single load-all operation if a change was detected in one or more key-values with refreshAll: true
-                                data = await LoadSelectedKeyValues(client, cancellationToken).ConfigureAwait(false);
+                                (data, orderedSelectedKeyValues) = await LoadSelectedKeyValues(client, cancellationToken).ConfigureAwait(false);
```

Comment:
We're doing this because there's no generic OrderedDictionary in .NET, however, I still feel that maintaining a new data structure to track the insertion order might be an overkill. It also adds the extra complication of having to return 2 objects from `LoadSelectedKeyValues`. 

Is it possible to convert `_mappedData` to `List<ConfigurationSetting>` instead of a dictionary? That would preserve the insertion order and we can still perform lookups because `ConfigurationSetting` already contains `key` and `label`.

Reply:
That's definitely fair, I considered making `_mappedData` a list but I was just concerned about having to iterate over the entire list for accessing/removing a specific key upon refresh. Then again, I don't really know how much of an impact that has.